src/components/grid/grid-container/grid-container.component.js

```diff
@@ -5,17 +5,43 @@ import GridItem from '../grid-item';
 
 const GridContainer = (props) => {
   const {
-    children
+    children,
+    m,
+    ml,
+    mr,
+    mt,
+    mb,
+    gridGap
   } = props;
   return (
-    <GridContainerStyle data-component='grid'>
+    <GridContainerStyle
+      data-component='grid'
+      m={ m }
+      ml={ ml }
+      mr={ mr }
+      mt={ mt }
+      mb={ mb }
+      gridGap={ gridGap }
+    >
       {children}
     </GridContainerStyle>
   );
 };
 
 GridContainer.propTypes = {
   /** Defines the Components to be rendered within the GridContainer. Requires a GridItem */
-  children: PropTypes.oneOfType([GridItem, PropTypes.arrayOf(GridItem)]).isRequired
+  children: PropTypes.oneOfType([GridItem, PropTypes.arrayOf(GridItem)]).isRequired,
+  /** Any valid CSS value to override default margin */
+  m: PropTypes.string,
```

Comment:
These can actually be numbers or strings. A number would be an index of the `space` array in the base theme config: https://github.com/Sage/carbon/blob/master/src/style/themes/base/base-theme.config.js